pre-alignment-qc-wf/main.nf

```diff
@@ -0,0 +1,242 @@
+
+download_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'max_retries': params.max_retries,
+    'first_retry_wait_time': params.first_retry_wait_time,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    *:(params.download ?: [:])
+]
+
+seqDataToLane_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'reads_max_discard_fraction': -1,
+    'publish_dir': params.publish_dir,
+    *:(params.seqDataToLane ?: [:])
+]
+
+payloadGen_params = [
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'publish_dir': params.publish_dir,
+    *:(params.payloadGen ?: [:])
+]
+
+uploadQc_params = [
+    'max_retries': params.max_retries,
+    'first_retry_wait_time': params.first_retry_wait_time,
+    'cpus': params.cpus,
+    'mem': params.mem,
+    'song_url': params.song_url,
+    'score_url': params.score_url,
+    'api_token': params.api_token,
+    *:(params.uploadQc ?: [:])
+]
+
+include { SongScoreDownload as dnld } from './wfpr_modules/github.com/icgc-argo/nextflow-data-processing-utility-tools/song-score-download@2.6.2/main.nf' params(download_params)
+include { payloadGenSeqExperiment as pGenExp } from './wfpr_modules/github.com/icgc-argo-workflows/data-processing-utility-tools/payload-gen-seq-experiment@0.5.0.1/main.nf' params(payloadGen_params)
+include { seqDataToLaneFastq as toLane } from './wfpr_modules/github.com/icgc-argo-workflows/dna-seq-processing-tools/seq-data-to-lane-fastq@0.2.0/main.nf' params(seqDataToLane_params)
+include { cutadapt } from './wfpr_modules/github.com/icgc-argo-workflows/argo-qc-tools/cutadapt@0.2.0/main.nf' params([*:params, 'cleanup': false])
+include { fastqc } from './wfpr_modules/github.com/icgc-argo-workflows/argo-qc-tools/fastqc@0.2.0/main.nf' params([*:params, 'cleanup': false])
+include { multiqc } from './wfpr_modules/github.com/icgc-argo-workflows/argo-qc-tools/multiqc@0.1.0/main.nf' params([*:params, 'cleanup': false])
+include { payloadGenQc as pGenQc } from './wfpr_modules/github.com/icgc-argo-workflows/data-processing-utility-tools/payload-gen-qc@0.1.0/main.nf' params(payloadGen_params)
+include { SongScoreUpload as upQc} from './wfpr_modules/github.com/icgc-argo/nextflow-data-processing-utility-tools/song-score-upload@2.7.0/main.nf' params(uploadQc_params)
+include { cleanupWorkdir as cleanup } from './wfpr_modules/github.com/icgc-argo-workflows/data-processing-utility-tools/cleanup-workdir@1.0.0.1/main.nf'
+
+
+// please update workflow code as needed
+workflow PreAlignmentQcWf {
+  take:  // update as needed
+    study_id
+    analysis_id
+    analysis_metadata
+    experiment_info_tsv
+    read_group_info_tsv
+    file_info_tsv
+    extra_info_tsv
+    sequencing_files
+
+
+  main:  // update as needed
+    // detect local mode or not
+    local_mode = false
+    if ((!analysis_metadata.startsWith("NO_FILE") || !experiment_info_tsv.startsWith("NO_FILE")) && sequencing_files.size() > 0){
```

Comment:
this logic could be re-written to be much clearer than two negatives and an ||. From what I'm reading, you could provide NO_FILE for either analysis_metadata or experiment info OR you could provide both of them as a real value and as long as the sequencing files are larger than zero, it will run in local mode. Is that the intended behavior?

Reply:
The logic is: 
`sequencing_files` is provided AND 
at least one of `analysis_metadata` or `experiment_info_tsv` is provided
It will run in local mode.

Reply:
Okay. It's confusing to read as is.